lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +999,17 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive_fmt not in ['zip', 'tar']:
+        archive_fmt = 'tar'
+
+    if archive:
+        cmd = "%s archive --format=%s --remote=%s --output=%s master" \
```

Comment:
git-archive does not work with http URLs, only git:// URLs.

If you run the command you get the following error:
`% git archive --format=tar --remote='https://github.com/ganeshrn/ansible.git' --output=/var/tmp/ansible master
fatal: Operation not supported by protocol.`

This is probably something that should be in the documentation, and checked for in the code.